polaris-react/src/styles/shared/_layout.scss

```diff
@@ -14,3 +14,32 @@
   #{$property}: calc(#{$spacing} + constant(safe-area-inset-#{$area}));
   #{$property}: calc(#{$spacing} + env(safe-area-inset-#{$area}));
 }
+
+@mixin custom-scrollbars() {
+  scrollbar-gutter: stable;
+
+  &::-webkit-scrollbar {
```

Comment:
I think we're going to encounter issues with supporting Firefox with this feature ([caniuse](https://caniuse.com/?search=%3A%3A-webkit-scrollbar))

"Non-standard: This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future."
https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar